api/src/pcapi/routes/backoffice/accounts/blueprint.py

```diff
@@ -1232,3 +1239,46 @@ def get_public_account_history(
     history = sorted(history, key=lambda item: item.actionDate or datetime.datetime.min, reverse=True)
 
     return history
+
+
+@public_accounts_blueprint.route("/<int:user_id>/gdpr-extract", methods=["POST"])
+@atomic()
+@utils.permission_required(perm_models.Permissions.EXTRACT_PUBLIC_ACCOUNT)
+def create_extract_user_gdpr_data(user_id: int) -> utils.BackofficeResponse:
+    user = (
+        users_models.User.query.filter(
+            users_models.User.id == user_id,
+            sa.or_(
+                users_models.User.roles.contains([users_models.UserRole.BENEFICIARY]),
+                users_models.User.roles.contains([users_models.UserRole.UNDERAGE_BENEFICIARY]),
+            ),
+        )
+        .options(sa.orm.joinedload(users_models.User.gdprUserDataExtract))
+        .one_or_none()
+    )
+    if not user:
+        raise NotFound()
+
+    if not is_valid_gdpr_user_extract(user=user):
+        flash("Une extraction de données est déjà en cours pour cet utilisateur.", "error")
+        return redirect(url_for(".get_public_account", user_id=user_id))
+
+    gdpr_data = users_models.GdprUserDataExtract(
+        userId=user_id,
+        authorUserId=current_user.id,
+    )
+    db.session.add(gdpr_data)
+    db.session.flush()
+
+    extract_beneficiary_data.delay(gdpr_tasks.ExtractBeneficiaryDataRequest(extract_id=gdpr_data.id))
```

Comment:
```suggestion
    on_commit(
        partial(
            extract_beneficiary_data.delay,
            payload=gdpr_tasks.ExtractBeneficiaryDataRequest(extract_id=gdpr_data.id),
        )
    )
```
tu auras besoin de `from pcapi.repository import on_commit` et de https://docs.python.org/fr/3/library/functools.html#functools.partial


explication:
La tu demandes a un autre serveur (connecté sur la même base de donnée) d'utiliser des données qui ne sont peut être pas encore arrivé dans la base de données (pas encore commit) donc tu vas avoir une course entre les deux serveur (race condition). Si le serveur de la coud task fait sa requête avant que le commit a lieu il ne trouvera pas la ligne et la tache sera juste ignorée. Le on_commit fait que la fonction qui lui est passé en argument (le partial) ne sera exécuté que après le commit donc la cloud task ne sera appelé que après que les données soient arrivées dans la base de données principale..